src/dawn/CodeGen/Cuda/MSCodeGen.cpp

```diff
@@ -717,6 +719,40 @@ void MSCodeGen::generateCudaKernelCode() {
   if(containsTemporary && useTmpIndex_)
     fnDecl = "template<typename TmpStorage>";
   fnDecl = fnDecl + "__global__ void";
+
+  int maxThreadsPerBlock =
+      blockSize_[0] * (blockSize_[1] + maxExtents[1].Plus - maxExtents[1].Minus +
+                       (maxExtents[0].Minus < 0 ? 1 : 0) + (maxExtents[0].Plus > 0 ? 1 : 0));
+
+  std::string arch = context_->getOptions().arch;
+  std::string domain_size = context_->getOptions().domain_size;
+  // if we have information about the architecture and the domain sizes we can further specify the
+  // minimum number of blocks per SM of the GPU
+  if((arch == "P100" || arch == "K40" || arch == "K80") && !domain_size.empty()) {
```

Comment:
should we not just be able to pass the SM's as an option-argument? This way we're less restricted

Reply:
I am not sure, yes, maybe, although this would leave all responsibility to the user in setting the right number...